src/main/java/lotto/service/LottoService.java

```diff
@@ -0,0 +1,51 @@
+package lotto.service;
+
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import lotto.Lotto;
+import lotto.LottoNumberGenerator;
+import lotto.LottoRank;
+import lotto.WinningLotto;
+import lotto.LottoResult;
+
+public class LottoService {
+    private final LottoNumberGenerator lottoNumberGenerator;
+
+    public LottoService(LottoNumberGenerator lottoNumberGenerator) {
+        this.lottoNumberGenerator = lottoNumberGenerator;
+    }
+
+    public List<Lotto> purchaseLottos(int purchaseAmount) {
+        int lottoCount = getLottoCount(purchaseAmount);
+
+        List<Lotto> lottos = new ArrayList<>();
+        for (int i = 0; i < lottoCount; i++) {
+            lottos.add(new Lotto(lottoNumberGenerator.generate()));
+        }
+        return lottos;
+    }
+
+    public int getLottoCount(int purchaseAmount) {
+        if (purchaseAmount % 1000 != 0) {
+            throw new IllegalArgumentException("[ERROR] 로또 구입 금액은 1,000원 단위여야 합니다.");
+        }
+        return purchaseAmount / 1000;
+    }
+
+    public LottoResult calculateResults(List<Lotto> lottos, WinningLotto winningLotto) {
```

Comment:
개인적으로 파라미터명을 lottos 말고 purchasedLottos처럼 구체적인 의미가 담긴 것으로 하는 것이 좋다고 생각합니다!
```suggestion
    public LottoResult calculateResults(List<Lotto> purchasedLottos, WinningLotto winningLotto) {
```